heron/metricsmgr/src/java/com/twitter/heron/metricsmgr/sink/WebSink.java

```diff
@@ -192,8 +195,20 @@ public void processRecord(MetricsRecord record) {
     if (isFlatMetrics) {
       metricsCache.putAll(processMetrics(source + "/", record.getMetrics()));
     } else {
-      Map<String, Double> sourceMap = processMetrics("", record.getMetrics());
-      metricsCache.put(source, sourceMap);
+      Map<String, Double> sourceCache;
+      Object sourceObj = metricsCache.getIfPresent(source);
+      if (sourceObj instanceof Map) {
+        @SuppressWarnings("unchecked")
+        Map<String, Double> castObj =  (Map<String, Double>) sourceObj;
+        sourceCache = castObj;
```

Comment:
Sadly we cannot. Apparently you can not annotate a definition if the variable is not also assigned.
e.g. 
```
@SuppressWarnings("unchecked")
Map<String, Double> sourceCache;
Object sourceObj = metricsCache.getIfPresent(source);
if (sourceObj instanceof Map) {
    sourceCache =  (Map<String, Double>) sourceObj;
...
```
breaks with `warning: [unchecked] unchecked cast` and
```
@SuppressWarnings("unchecked")
sourceCache =  (Map<String, Double>) sourceObj;
```
is not legal syntax